cobalt/build/android/package.py

```diff
@@ -30,15 +30,45 @@
 
 import argparse
 import subprocess
+import sys
+import os
 
 PLATFORMS = ['android-arm', 'android-arm64', 'android-x86']
 CONFIGS = ['gold', 'qa']
 
 
+def patch_args_and_regen(out_dir):
+  """
+  Ensures 'is_cobalt_on_google3 = true' is in the args.gn file and re-runs
+  'gn gen' to apply the change.
+  """
+  args_gn_path = os.path.join(out_dir, 'args.gn')
+
+  try:
+    with open(args_gn_path, 'r+', encoding='utf-8') as f:
+      if not any(line.strip().startswith('is_cobalt_on_google3') for line in f):
+        # The argument was not found, append it, ensuring a newline.
+        f.write('\nis_cobalt_on_google3 = true\n')
+  except FileNotFoundError:
```

Comment:
What's the reason we want to catch errors and call sys.exit(1) instead of just using the error originally raised? If you do want to do it this way, instead use `os.path.exists` with a print and return instead of a try/except